docs/components/data_tables.md

```diff
@@ -0,0 +1,386 @@
+      _Row('Cell A3', 'CellB3', 'CellC3', 3),
+      _Row('Cell A4', 'CellB4', 'CellC4', 4),
+    ];
+  }
+
+  final BuildContext context;
+  List<_Row> _rows;
+
+  int _selectedCount = 0;
+
+  @override
+  DataRow getRow(int index) {
+    assert(index >= 0);
+    if (index >= _rows.length) return null;
+    final row = _rows[index];
+    return DataRow.byIndex(
+      index: index,
+      selected: row.selected,
+      onSelectChanged: (value) {
+        if (row.selected != value) {
+          _selectedCount += value ? 1 : -1;
+          assert(_selectedCount >= 0);
+          row.selected = value;
+          notifyListeners();
+        }
+      },
+      cells: [
+        DataCell(Text(row.valueA)),
+        DataCell(Text(row.valueB)),
+        DataCell(Text(row.valueC)),
+        DataCell(Text(row.valueD.toString())),
+      ],
+    );
+  }
+
+  @override
+  int get rowCount => _rows.length;
+
+  @override
+  bool get isRowCountApproximate => false;
+
+  @override
+  int get selectedRowCount => _selectedCount;
+}
+
+```
+
+### Key properties
+
+The following is an anatomy diagram of a data table:
+
+![Data table anatomy diagram](assets/data_tables/data-table-anatomy.png)
+
+A data table consists of the following properties:
+
+1. Container
+1. Column header names
+1. Sort button
+1. Row checkboxes
+1. Rows
+
+### Container
+
+| &nbsp; | Property |
+| --- | --- |
+| **Color** | style |
+| **Stroke color** | while setting `ThemeData` fill out `divideColor` parameter |
+
+### Column header names
+
+| &nbsp; | Property |
+| --- | --- |
+| **Label text** | Use `columns`, within `Column` set create `DataColumn` widget fill out `label` property with `Text` Widget|
+| **Typography** | Within the `Text` widget from note above fill out `style` property with `TextStyle` widget|
+| **Color** | Within the `Text` widget from note above fill out `style` property with `TextStyle` widget fill out the `color` property |
+
+### Sort button
+
+Sorting button can't be customized.
+
+### Row checkboxes
+
+| &nbsp; | Property |
+| --- | --- |
+| **active Color** | while setting `ThemeData` fill out `accentColor`|
+
+### Rows
+
+| &nbsp; | Property |
+| --- | --- |
+| **Label text** | Use `DataCell` widget, fill out with property with `Text` Widget |
+| **Typography** | Within the `Text` widget from note above fill out `style` property with `TextStyle` widget|
+| **Color** |  Within the `Text` widget from note above fill out `style` property with `TextStyle` widget fill out the `color` property  |
+
+## Theming data tables
+
+Data Table cells children are Widgets. This means that the Data table is dependent on the Widgets that the user puts within it. The only theme variables that affect the DataTable is:
+
+* dividerColor
+* secondaryHeaderColor
+* accentColor
+
+[Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/data_tables/normal)
+
```

Comment:
Include an introductory sentence, for example:

"The following is an example data table using the [Shrine theme](https://material.io/design/material-studies/shrine.html)."

Reply:
Done